app/helpers/three_column_data_helper.rb

```diff
@@ -1,22 +1,11 @@
 module ThreeColumnDataHelper
   def split_data_into_three_columns(data_set)
-    if data_set.size > 1
-      # Divide the total by three and round up if possible to make sure the data is evenly distributed amongst the columns
-      count = ((data_set.size * 2) + 3) / (3 * 2)
-      sliced_data_set = data_set.each_slice(count).to_a
-
-      # When the data set count is divisible by 3 with a remainder of 1, move item(s) from one array to the previous to maintain distribution
-      if data_set.size % 3 == 1
-        sliced_data_set[2].insert(-1, sliced_data_set[3][0])
-        sliced_data_set[1].insert(-1, sliced_data_set[2][0], sliced_data_set[2][1])
-        sliced_data_set[0].insert(-1, sliced_data_set[1][0])
-        sliced_data_set[2].shift(2)
-        sliced_data_set[1].shift
-      end
-
-      sliced_data_set
-    else
-      data_set
+    flat_data_set = data_set.to_a.flatten
+    count_per_column = (flat_data_set.size + 2) / 3
+    sliced_data_set = Array.new(3) { |i| flat_data_set[i * count_per_column, count_per_column] || [] }
+    if flat_data_set.size % 3
+      sliced_data_set[1] << sliced_data_set[2].shift if sliced_data_set[2].any?
     end
+    sliced_data_set
   end
-end
\ No newline at end of file
+end
```

Comment:
Had to edit this method in order to work with the difficult set up in the innovation editor intro view, went ahead and refactored for readability and robustness while I was at it. Should work just the same as before for existing category rendering in both the innovation editor intro view as well as the search filters.